StudyConnect.API/Controllers/Forum/PostController.cs

```diff
@@ -8,84 +13,209 @@ namespace StudyConnect.API.Controllers.Forum;
+
+        if (posts.Data == null)
+            return NotFound("Posts not found.");
+
+        var result = posts.Data.Select(p => GeneratePostDto(p));
+
+        return Ok(result);
     }
 
     /// <summary>
     /// Get a post by its ID
     /// </summary>
-    /// <param name="pid"> unique identifier of the post </param>
-    /// <returns> the post details in JSON </returns>
-    [HttpGet("{pid}")]
-    [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
-    public IActionResult GetPost([FromRoute] Guid pid ) 
+    /// <param name="pid">The unique identifier of the post.</param>
+    /// <returns>On success a Dto with information about the post, on failure HTTP 400/404 status code.</returns>
+    [HttpGet("{pid:guid}")]
+    public async Task<IActionResult> GetPostById([FromRoute] Guid pid)
     {
-        string mockPost = @"
-        { 
-            ""PostID"": ""d2b876f0-d6h9-4a02-8965-5d248b573j8l"",
-            ""Author"": ""John Doe"",  
-            ""Title"": ""Mock Discussion Post Title"",
-            ""Tags"": [""C#"", ""API"", ""Mocking""],
-            ""MadeAt"": ""2025-03-29T12:34:56"",
-            ""Comments"": [
-                {
-                    ""CommentID"": ""d2b516f0-d3f5-4a02-8191-5d122c375b2d"",
-                    ""PostID"": ""d2b876f0-d6h9-4a02-8965-5d248b573j8l"",
-                    ""Author"": ""John Doe"",
-                    ""Content"": ""This is a mock content for a mock post."",
-                    ""MadeAt"": ""2025-03-29T12:34:56""
-                }
-
-            ]
-        }";
-
-        return Ok(mockPost);
+        var result = await _postRepository.GetByIdAsync(pid);
+        if (!result.IsSuccess)
+            return BadRequest(result.ErrorMessage);
+
+        if (result.Data == null)
+            return NotFound("Post not found.");
+
+        var postDto = GeneratePostDto(result.Data);
+
+        return Ok(postDto);
     }
 
     /// <summary>
     /// Update existing post
     /// </summary>
     /// <param name="pid"> unique identifier of the post </param>
-    /// <returns> HTTP 200 OK response on success </returns>
-    [HttpPut("{pid}")]
-    public IActionResult UpdatePost([FromRoute] Guid pid)
+    /// <param name="updateDto"> a dto containing the data for updating the post. </param>
+    /// <returns>On success a HTTP 200 status code, on failure a HTTP 400 status code.</returns>
+    [HttpPut("{pid:guid}")]
+    public async Task<IActionResult> UpdatePost([FromRoute] Guid pid, [FromBody] PostUpdateDto updateDto)
     {
-        return Ok();
+        if (!ModelState.IsValid)
+            return BadRequest(ModelState);
+
+        var post = new ForumPost
+        {
+            Title = updateDto.Title,
+            Content = updateDto.Content
+        };
+
+        var result = await _postRepository.UpdateAsync(pid, post);
+        if (!result.IsSuccess)
+            return BadRequest(result.ErrorMessage);
+
+        if (result.Data == false)
```

Comment:
```suggestion
        if (!result.Data)
```
SonarQube suggestion